tests/core/test__docs_decorators.py

```diff
@@ -0,0 +1,586 @@
+import pytest
+
+from great_expectations.core._docs_decorators import (
+    deprecated_argument,
+    deprecated_method,
+    new_argument,
+    new_method,
+    public_api,
+)
+
+# @public_api
+
+
+@public_api
+def _func_full_docstring_public_api(some_arg, other_arg):
+    """My docstring.
+
+    Longer description.
+
+    Args:
+        some_arg: describe some_arg
+        other_arg: describe other_arg
+    """
+    pass
```

Comment:
placeholders?

Reply:
Ah, these are more like test fixtures. They are functions and classes with the appropriate decorator (e.g. `@public_api` in this case). We then check to make sure their docstrings are modified appropriately. I didn't want to mark as a pytest fixture to make sure there is no conflicting processing that happens that might impact the docstring processing.